test/src/P528GTestUtils.cpp

```diff
@@ -0,0 +1,194 @@
+    #endif
+}
+
+std::string GetDirectory(std::string name) {
+    std::string dataDir(__FILE__);
+    dataDir.resize(dataDir.find_last_of("/\\"));
+    dataDir.resize(dataDir.find_last_of("/\\"));
+    AppendDirectorySep(dataDir);
+    dataDir += name;
+    AppendDirectorySep(dataDir);
+    return dataDir;
+}
+
+std::vector<InputsAndResult> ReadP528InputsAndResultFromPoints(const std::string& filename) {
+    std::vector<InputsAndResult> testData;
+    std::string dataDir = GetDirectory("data");
+    std::ifstream file(dataDir + filename);
+    std::string line;
+    InputsAndResult d;  // struct to store data from a single line of CSV
+    char c;  // single-character representing the comma (delimiter)
+    while (std::getline(file, line)) {
+        std::istringstream iss(line);
+        if (iss >> d.d__km >> c >> d.h_1__meter >> c >> d.h_2__meter >> c >> d.f__mhz >> c >> d.T_pol >> c >> d.p 
+            >> c >> d.expectedReturn >> c >> d.expectedResult.propagation_mode >> c >> d.expectedResult.warnings >> c >> d.expectedResult.d__km
+            >> c >> d.expectedResult.A__db >> c >> d.expectedResult.A_fs__db >> c >> d.expectedResult.A_a__db >> c >> d.expectedResult.theta_h1__rad
+        ) {
+            testData.push_back(d);
+        }
+    }
+    return testData;
+}
+
+std::vector<std::string> GetDataTablesFileList()
+{
+    std::vector<std::string> ret;
+    std::string dataTableDir = GetDirectory("Data Tables");
+    for (const auto& entry : std::filesystem::directory_iterator(dataTableDir)) {
+        //std::cout << entry.path() << std::endl;
+        //std::string filename = entry.path().filename();
+        ret.push_back(entry.path().filename().string());
+    }
+    return ret;
+}
+
+std::vector<InputsAndResult> ReadP528InputsAndResultFromDataTable(const std::string& filename, int testStep) {
+    std::vector<InputsAndResult> testData;
+    std::string dataDir = GetDirectory("Data Tables");
+    std::ifstream file(dataDir + filename);
+    InputsAndResult d;  // struct to store data from a single line of CSV
+
+    double d__km;
+    std::vector<double> h_1__meter;
+    std::vector<double> h_2__meter;
+    double f__mhz;
+    int T_pol = 0;
+    double p;
+
+    //std::vector<double> A__db;
+    double A_fs__db;
+
+    std::vector<std::vector<std::string>> csvRows = readCSV(file);
+    if (csvRows.size() <= 4) {
+        return testData;
+    }
+
+    std::vector<std::string> cells;
+   
+    if (csvRows[0].size() > 0) {
+        //std::cout << csvRows[0][0] << std::endl;
+        string strCell = csvRows[0][0];
+        strCell.resize(strCell.find("M"));
+        f__mhz = std::stod(strCell);
+        strCell = csvRows[0][0];
+        strCell.resize(strCell.find(")"));
+        strCell = strCell.substr(strCell.find("(") + 1);
+        p = std::stod(strCell) * 100.0;
+    }
+    if (csvRows[1].size() > 2) {
+        //std::cout << csvRows[1][2] << std::endl;
+        for (int i = 2; i < csvRows[1].size(); i++) {
```

Comment:
Using `int` here causes a mismatch between datatype of `i` and the value returned by `csvRows[1].size()` and the compiler complains when using the `-Wall` option. This can be avoided using the same type that is returned by the `size()` function. Its type is long so `typedef`ing can help declutter the code. This issue also happens in other `for` loops in this file. 

```suggestion
        typedef std::vector<std::vector<std::string> >::size_type vec_size_t
        for (vec_size_t i = 2; i < csvRows[1].size(); i++) {
```